ddtrace/tracer/rules_sampler.go

```diff
@@ -154,17 +182,43 @@ func RateRule(rate float64) SamplingRule {
 	}
 }
 
+// TagsResourceRule returns a SamplingRule that applies the provided sampling rate to traces with spans that match
+// resource, name, service and tags provided. Additionally, targetRoot specifies whether the rule should apply to root spans only.
+func TagsResourceRule(tags map[string]*regexp.Regexp, resource, name, service string, rate float64, targetRoot bool) SamplingRule {
+	return SamplingRule{
+		Service:    globMatch(service),
+		Name:       globMatch(name),
+		Resource:   globMatch(resource),
+		Rate:       rate,
+		TargetRoot: targetRoot,
+		Tags:       tags,
+		ruleType:   SamplingRuleTrace,
+	}
+}
+
+// SpanTagsResourceRule returns a SamplingRule that applies the provided sampling rate to spans that match
+// resource, name, service and tags provided.
+func SpanTagsResourceRule(tags map[string]*regexp.Regexp, resource, name, service string, rate float64) SamplingRule {
```

Comment:
Rather than requiring someone to pass in a regular expression for the tags, can we let them pass in a map[string]string with a glob pattern that we then convert to the necessary regexp type? I think that'll be a better UX for them. 
(Described a bit here - https://datadoghq.atlassian.net/browse/AIT-9102?focusedCommentId=1466227)